### ISSUE TITLE
The training code

### ISSUE THREAD

**Issue by ForrestPi** [zqbai-jeremy/DFNRMVS#3]
Could you provide the training part codes？
Thanks

**Comment by zqbai-jeremy** [comment]
The training code is quite messy thus not easy to clean and use. I can provide loss functions and training parameters shortly. The rest is just normal training pipeline.

**Comment by ForrestPi** [comment]
@zqbai-jeremy 
Thanks, wish for the loss functions and training parameters

**Comment by zqbai-jeremy** [comment]
@ForrestPi 
The loss functions and training parameters have been uploaded. Please let me know if you have any further questions. Thank you for your interest in our work.

**Comment by moranli19** [comment]
Is there any code or more details for training data processing?  

The StirlingESRS_3Dface dataset contains 74 females and 64 different male subjects. Each subject with 8 expressions. As you describe in the paper 'Then, we render one image for each expression with different poses and same global illumination using Spherical Harmonics (SH).', the amount of training sample should be (74+64)*C(8,2) = 3864 samples. This number does not conform to 8K as you mentioned 'As a result, around 8K training samples are generated.'
Is there any misunderstanding? 

Thanks for any reply~!

**Comment by zqbai-jeremy** [comment]
@moranli19 
For each identity, we generate 100 samples with randomly sampled 2 expressions, 2 poses, and 1 illumination. Then we detect landmarks on rendered images and check whether they are consistent with landmarks on 3D scans (since the landmark detector and the NICP registration may have errors). If not, the sample is discard. As a result, we finally have around 8K samples.
Please let me if you have any further questions. Thank you!
